vagrant/bin/change-timeout.sh

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+set -e
+
+## Default values
+RESET=
+TIMEOUT_LENGTH=3600
+
+## Verify pre-requisites
+if [[ ! -f /etc/.vagranthost ]]; then
+    >&2 echo "Error: This script should be run from the host machine."
+    exit 1
+fi
+
+## Argument parsing
+for arg in "$@"; do
+    case $arg in
+        --length=*)
+            TIMEOUT_LENGTH="${arg#*=}"
+            if [[ ! "${TIMEOUT_LENGTH}" =~ ^[0-9]+$ ]]; then
+                >&2 echo "Error: Invalid length given --length=${TIMEOUT_LENGTH}"
+                exit -1
+            fi
+            ;;
+        -r|--reset)
+            RESET=1
+            ;;
+        --help)
+            echo "Usage: $(basename $0) [--length=<timeout in seconds>] [--reset]"
+            echo "Change MySQL wait_timeout, PHP mysql.connect_timeout and default_socket_timeout, and Nginx proxy_read_timeout values."
+            echo "Useful for long-running PHP debugging sessions."
+            echo ""
+            echo "       --length       Timeout length in seconds (defaults to ${TIMEOUT_LENGTH})"
+            echo "  -r : --reset        Reset timeouts to system defaults"
+            echo ""
+            exit -1
+            ;;
+        *)
+            >&2 echo "Error: Unrecognized argument $arg"
+            exit -1
+            ;;
+    esac
+done
+
+vagrant ssh web -- "
```

Comment:
Would you change these to wrap the script argument in single-quotes? This will remove the need to escape everything, including the dollar-signs which are actually not currently escaped, only escape-avoided because of the brackets.